lib/chewy/search/response.rb

```diff
@@ -33,6 +39,17 @@ def max_score
         @max_score ||= hits_root['max_score']
       end
 
+      # Response `errors` field. Returns `nil` if there is no error.
+      # @return [Array, nil]
+      def errors
+        errors = Array(@body['errors'])
+        # there's another case when failures could be present in some shards, in that case es returns them
+        # in @body[_shards][failures] array
+        errors += Array(@body['_shards']['failures']) if @body['_shards']
```

Comment:
Can you categorise these seperately under a key ? So that we can handle them seperately if needed in leadgenie. 
Also we can start sending error type in the logs

Reply:
Nice idea.

I'll change the return type of `errors` method to hash, something like:
```ruby
{
    "errors": Array(@body['errors']),
    "shards_failures": Array(@body['_shards']['failures']),
}
```
wdyt?

Reply:
Sure, this works yeah. Or it can be a seperate key within the errors itself with a different 'type'. Either way is fine